### PR TITLE
Mm 62229 schmsg not removed on reconnect

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR is intended to resync the scheduled messages information on websocket reconection.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-62229

#### Screenshots

In the video, you see how the scheduled messages list looks after a laptop wakeup that during the iddle time, the messages were sent.
Note: in order to notice the effect of the code in this PR, there was a timeout of 30sec that was removed here https://github.com/mattermost/mattermost/pull/29623/commits/2e3a7478a908756cb50beeeb567b2f37d4cf6fb6

https://github.com/user-attachments/assets/bf55b112-9a25-4511-9164-432d08d65cea

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/actions/websocket_actions.jsx [hunk 1]
```diff
@@ -239,6 +241,7 @@ export function reconnect() {
         }
 
         dispatch(fetchAllMyTeamsChannels());
+        dispatch(fetchTeamScheduledPosts(currentTeamId, true));
```
Comment (pvev) [inline webapp/channels/src/actions/websocket_actions.jsx:244]: this is the actual change in order to get the scheduled messages slice refreshed on ws reconnect. The other part is for adding the time to the logged ws events information.

**Review by harshilsharma63 — APPROVED** [review]

File: webapp/channels/src/actions/websocket_actions.jsx (hunk 1 above)
Comment (hmhealey) [inline webapp/channels/src/actions/websocket_actions.jsx:244]: This only resyncs for the current team. Will the other teams be resynced when we switch to them?

File: webapp/channels/src/actions/websocket_actions.jsx [hunk 2]
```diff
@@ -217,7 +219,7 @@ function restart() {
 
 export function reconnect() {
     // eslint-disable-next-line
-    console.log('Reconnecting WebSocket');
+    console.log(`Reconnecting WebSocket - ${getCurrentTime()}`); //eslint-disable-line no-console
```
Comment (hmhealey) [inline webapp/channels/src/actions/websocket_actions.jsx:222]: I'm totally fine leaving this, but Chrome and Firefox (maybe Safari too, but I never checked it) both have options to show timestamps for every message in the JS console so we might not actually need this

File: webapp/channels/src/actions/websocket_actions.jsx (hunk 1 above)
Comment (pvev) [reply webapp/channels/src/actions/websocket_actions.jsx:244]: yes, on every team switch, the scheduled messages information is resynced.

File: webapp/channels/src/actions/websocket_actions.jsx (hunk 2 above)
Comment (pvev) [reply webapp/channels/src/actions/websocket_actions.jsx:222]: you are right @hmhealey , it even show the miliseconds, was not aware of it, will remove this then 👍 
Screenshot 2024-12-18 at 10 30 36 AM

**Review by marianunez — APPROVED** [review]
Stepping in to do a QA pass as Yasser is OOO. Schedule message is removed from the list when reconnecting after sent. I still see https://mattermost.atlassian.net/browse/MM-62222 but seems that will be fixed after this is merged with https://github.com/mattermost/mattermost/pull/29604

Yasser can do a post-merge validation at release time.

**Comment by marianunez** [comment]
@pvev same as https://github.com/mattermost/mattermost/pull/29604 we should cherry pick this to v10.4

**Comment by pvev** [comment]
@amyblais can we please cherry pick this to v10.4?

**Comment by marianunez** [comment]
> @amyblais can we please cherry pick this to v10.4?

@pvev Amy is OOO. You can try the `/cherry-pick` command or do it manually to the release-10.4 branch

**Pull request merged**